File: typescript/src/get_aggregated_data.ts
import { createNodeClient } from "./client"
import { create, toJsonString } from "@bufbuild/protobuf";
import {
  AggregatedDataRequestSchema,
  AggregatedResponseSchema,
  AggregateVariant,
  VehicleGroupBy
} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb";
import { TimeUnit } from "@buf/compassiot_model.bufbuild_es/compassiot/compass/v1/time_pb";

const client = createNodeClient()

const request = create(AggregatedDataRequestSchema, {
  aggregateVariant: AggregateVariant.COST,
  // if vins is empty, aggregates over all vehicles in the workspace
  vins: ["LRW3F7FS9PC893577"],
  dateTimeRange: {
    start: { day: 1, month: 5, year: 2026 },
    end: { day: 31, month: 5, year: 2026 }
  },
  // GroupBy
  granularity: TimeUnit.DAY,
  vehicleGroupBy: VehicleGroupBy.VIN
})

const response = await client.getAggregatedData(request)

console.log(toJsonString(AggregatedResponseSchema, response, { prettySpaces: 2 }))

File: python/get_aggregated_data.py
from client import create_native_client
import compassiot.compass.v1.time_pb2 as time
import nativeconnect.api.v1.nativeconnect_pb2 as native


def main():
    client = create_native_client()

    request = native.AggregatedDataRequest(
        aggregate_variant=native.AggregateVariant.AGGREGATE_VARIANT_COST,
        # if vins is empty, aggregates over all vehicles in the workspace
        vins=["LRW3F7FS9PC893577"],
        date_time_range=time.DateTimeRange(
            start=time.LocalDate(day=1, month=5, year=2026),
            end=time.LocalDate(day=31, month=5, year=2026)
        ),
        # GroupBy
        granularity=time.TimeUnit.DAY,
        vehicle_group_by=native.VehicleGroupBy.VEHICLE_GROUP_BY_VIN
    )

    response = client.GetAggregatedData(request)

    print(response)


if __name__ == "__main__":
    main()
